.github/workflows/workflow-build-push-container-github-registry.yml

```diff
@@ -0,0 +1,124 @@
+name: Reusable workflow to build and push docker container to github container registry
+
+on:
+  workflow_call:
+    inputs:
+      container-name:
+        required: true
+        type: string
+      tag:
+        required: true
+        type: string
+      registry:
+        required: true
+        type: string
+  pull_request:
+    types:
+      - opened
+      - closed
+      - synchronize
+
+jobs:
+  build-push-image:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Out Repo
+        uses: actions/checkout@v4
+
+      - name: Check if a Dockerfile is present at the root of the repo
+        id: check-file
+        run: |
+          if [ -f Dockerfile ]; then
+            echo "::set-output name=exists::true"
+          else
+            echo "::set-output name=exists::false"
+          fi
+
+      - name: Cancel the workflow if no Dockerfile is present
```

Comment:
don't accept having an asynchronous operation and then waiting an arbitrary amount of time for the operation to complete (without validating it's done).

There is a way to check:

https://stackoverflow.com/questions/60589373/how-to-force-job-to-exit-in-github-actions-step

The exit behavior can be achieved with gh run cancel and gh run watch commands:

- name: Early exit
  run: |
    gh run cancel ${{ github.run_id }}
    gh run watch ${{ github.run_id }}
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
The watch is required since cancellation will not abort immediately.

You may need actions: 'write' permission added for the job, something like:

permissions:
  ...
  actions: 'write'


OR you can make every subsequent step conditional to the presence or not of the Dockerfile with if: